### ISSUE TITLE
[Question] Parameterized @observable decorator?

### ISSUE THREAD

**Issue by rzcoder** [mobxjs/mobx#660]
Hey!

If I need observable property (it's an object) be observable only by reference, I'd like to write something like:

```js
class State {
    @observable({asReference: true}) page;
}
```

but instead I have to write

```js
class State {
    constructor() {
        extendObservable(this, {
            page: asReference(null)
        });
    }
}
```

So, what you think about first variant?

**Comment by mweststrate** [comment]
See also #649, we are rethinking additional api's on how observables can be constructed.

Note that `@observable page = asReference(null)` should also work.

**Comment by mweststrate** [comment]
Closed in favor of #649
